dhis-2/dhis-services/dhis-service-core/src/main/java/org/hisp/dhis/program/function/D2HasValue.java

```diff
@@ -44,7 +45,9 @@
     @Override
     public Object getDescription( ExprContext ctx, CommonExpressionVisitor visitor )
     {
-        return getProgramArgType( ctx ).getDescription( ctx, visitor );
+        getProgramArgType( ctx ).getDescription( ctx, visitor );
+
+        return DEFAULT_BOOLEAN_VALUE;
```

Comment:
Could you explain the logic here, i.e. we always return the default?

Reply:
`getDescription() `function is being used for expression validation and the way it works is all the variables and d2 function are replaced by default values and then the expression is evaluated.

Reply:
@jimgrace ?

Reply:
Zubair's fix looks right to me. The `getDescription()` method builds a Map of string substitutions to turn an expression into an expression description, for example replace "#{fbfJHSPpUQD}" with "ANC 1st visit". But what the method *returns* is a dummy value so the overall expression can be syntax-checked.

The ANTLR parsing methods include some type checking. So for example, the first argument to an `if()` function should be boolean, as in `if( test, valueIfTrue, valueIfFalse )`. A program indicator filter should also evaluate to a boolean. The `D2:HasValue()` function should always return a dummy boolean. It was incorrectly trying to return the datatype of the value it was testing.

Reply:
Thanks for the feedback.